pages/noun/[nounId].tsx

```diff
@@ -0,0 +1,55 @@
+import Auction from '../../components/Auction';
+import SiteHead from '../../components/SiteHead';
+import SiteHeader from '../../components/SiteHeader';
+import SiteFooter from '../../components/SiteFooter';
+import { GetServerSideProps } from 'next';
+import getAuctionData from '../../server/getAuctionData';
+import { AuctionData, Wallet } from '../../server/api/types';
+
+export default function NounAuctionPage({ auctionData }: { auctionData: AuctionData }) {
+  const lookup: { [address: string]: Wallet } = Object.fromEntries(
+    auctionData.wallets.map((wallet) => [wallet.address, wallet]),
+  );
+
+  const data = {
+    nounId: auctionData.auction.id,
+    startTime: auctionData.auction.startTime,
+    winnerAddress: auctionData.auction.winner,
+    winnerENS: lookup[auctionData.auction.winner ?? '']?.ens || null,
```

Comment:
you don't need to construct `lookup` here, simply do this, since you only need one value

```suggestion
    winnerENS: auctionData.wallets.find(wallet => wallet.address === auctionData.auction.winner) || null,
```